src/highlevel.jl

```diff
@@ -61,21 +60,20 @@ Fields:
 """
 mutable struct Device
     ptr::Ptr{SoapySDRDevice}
-    function Device(dev_ptr)
+    function Device(args)
+        dev_ptr = SoapySDRDevice_make(args)
         if dev_ptr == C_NULL
             throw(ArgumentError("Unable to open device!"))
         end
-        return new(dev_ptr)
+        this = new(dev_ptr)
+        finalizer(this) do this
+            SoapySDRDevice_unmake(this.ptr)
+        end
```

Comment:
We should still support `close(::Device)`, set `ptr` to `C_NULL` within `close()`, and wrap this call in `if this.ptr != C_NULL` so that users can eagerly close devices.

Reply:
Ugh okay, on second though, we probably shouldn't allow eager `close` at all for simplicity. If we allow `close` we need to use null pointer checks in pretty much all functions (`setproperty`, `getproperty`, etc...) . On the otherhand by not allowing eager `close`, we just get `UndefVarError` for free without this complexity.